### ISSUE TITLE
feat: expose shelf's additional Response constructors for better DevUX

### ISSUE THREAD

**Issue by erickzanardo** [VeryGoodOpenSource/dart_frog#594]
**Description**

Although not needed, helpers from shelf like `Response.movedPermanetly` are good shortcuts that dart frog could also expose in its `Response` class.

**Requirements**

- [ ] Expose the methods from [shelf](https://pub.dev/documentation/shelf/latest/shelf/Response-class.html) in Dart Frog's Response class

**Comment by felangel** [comment]
In my opinion it’s a bit weird to have named constructors for both the content type as well as the status code. All the named constructors dart frog currently has are for configuring the content type (Response.json, etc) but if we add named constructors for customizing status code then it starts to become confusing because what if I want to return a 400 bad request with some json? I can do that either via Response.json(statusCode: 400, payload) or Response.badRequest(headers: …, body: json.encode(payload)). Now we have two different ways of doing the exact same thing which is not ideal imo.

**Comment by erickzanardo** [comment]
That is a fair point @felangel, but a counter point IMO is the `Response.movedPermanetly`, that type of response is composed of the status code and an specific header, it shouldn't have body (I mean, you probably could have body, but it makes no sense AFAIK to have one in that kind of  response). And it makes a nicer helper to have a `Response.movedPermanently('url')` instead of having to always write

```
Response(
  status: 301,
  headers: { 'Location': 'url' },
)
```

I can see your point and partially agree with it, maybe the badRequest one is a contructor that shouldn't exist, but I believe some of them like the `movedPermantely` one, should, does that makes sense to you @felangel ?

**Comment by felangel** [comment]
@erickzanardo yeah I agree movedPermanently would be convenient and I have less of a problem with that one but I think it’ll also open the door for others to request statusCode related named constructors (badRequest, unauthorized, forbidden) since dart frog will have set a precedent.

**Comment by erickzanardo** [comment]
@felangel can't we simply not write the ones that we feel will bring more harm than good? IMO, that door is already open since we have `Response.json`, which is nothing more than a helper too?

**Comment by Spiritcow** [comment]
@erickzanardo @felangel,  so should I delete all constructors from my PR except movedPermantely?

**Comment by AyadLaouissi** [comment]
I can work on this task

**Comment by alestiago** [comment]
I'm closing since move permanently has been merged. 

If there are any other constructors that are seen as valuable feel free to re-open or open a new issue for the specific constructor.
